server/src/test/java/org/elasticsearch/index/mapper/AbstractScriptFieldTypeTestCase.java

```diff
@@ -166,21 +166,38 @@ public void testFieldCaps() throws Exception {
     }
 
     /**
-     * Check that running query on a runtime field that throws errors with enabled lenient error handling catches those silently
+     * Check that running query on a runtime field script that fails has the expected behaviour according to its configuration
      */
-    public void testQueryErrorHandling() throws IOException {
+    public final void testOnScriptError() throws IOException {
         try (Directory directory = newDirectory(); RandomIndexWriter iw = new RandomIndexWriter(random(), directory)) {
             iw.addDocument(List.of(new StoredField("_source", new BytesRef("{\"foo\": [1]}"))));
             try (DirectoryReader reader = iw.getReader()) {
                 IndexSearcher searcher = newUnthreadedSearcher(reader);
-                AbstractScriptFieldType<?> fieldType = build("error", Collections.emptyMap(), OnScriptError.CONTINUE);
+                {
+                    AbstractScriptFieldType<?> fieldType = build("error", Collections.emptyMap(), OnScriptError.CONTINUE);
+                    SearchExecutionContext searchExecutionContext = mockContext(true, fieldType);
+                    Query query = new ExistsQueryBuilder("test").rewrite(searchExecutionContext).toQuery(searchExecutionContext);
+                    assertEquals(0, searcher.count(query));
+                }
+                {
+                    AbstractScriptFieldType<?> fieldType = build("error", Collections.emptyMap(), OnScriptError.FAIL);
+                    SearchExecutionContext searchExecutionContext = mockContext(true, fieldType);
+                    Query query = new ExistsQueryBuilder("test").rewrite(searchExecutionContext).toQuery(searchExecutionContext);
+                    expectThrows(RuntimeException.class, () -> searcher.count(query));
+                }
+            }
+        }
+    }
+
+    public final void testOnScriptErrorFail() throws IOException {
```

Comment:
Just looking at the merged changes - I'm trying to understand why this test isn't covered by the second code block in `testOnScriptError()`, looks identical to me so far. What am I missing?

Reply:
you are right, I am not sure what I was trying to do. It was last year :)  I may have wanted to remove that second block in favour of the separate test.